system-tests/test.py

```diff
@@ -47,8 +49,13 @@ def teardown(self):
         shutil.rmtree(str(working_dir))
 
     def run(self):
-        command = ['python3', '-m', 'fab', self._test_directory]
-        environment = {'PYTHONPATH': 'source'}
+        command = ['python3', '-m', 'fab']
+        if self._fpp_flags is not None:
+            command.append('--fpp-flags=' + self._fpp_flags)
+        command.append(self._test_directory)
+
+        environment = os.environ
+        environment.update({'PYTHONPATH': 'source'})
```

Comment:
I don't think the wholesale copying of the users environment is a good way to go. It risks pulling in a lot of complicating factors. Instead please try copying only the required variables across. e.g. PATH. It may be necessary to extend rather than replace PYTHONPATH in the future.

Reply:
Okay, I'll give it a try